### PR TITLE
Allow data to be an array

### LINKED ISSUE #64: Support array objects as data
### Steps to reproduce

```javascript
const data = [
    {
        hello: "world"
    }
];
throw new errors.BadRequest("Input not valid", data);
```
### Expected behavior
Response: 

```json
{
    "name": "BadRequest",
     "message": "JSON-Schema validation failed",
     "code": 400,
     "className": "bad-request",
     "data": [
        {
            "hello": "world"
        }
     ],
    "errors":  { }
}
```

### Actual behavior
Response:

```json
{
    "name": "BadRequest",
     "message": "JSON-Schema validation failed",
     "code": 400,
     "className": "bad-request",
     "data": {
        "0": {
            "hello": "world"
        }
     },
    "errors":  { }
}
```

### More
The problem is here:
https://github.com/feathersjs/feathers-errors/blob/master/src/index.js#L36

### Proposed solution
If we could make the 'errors' property of the data object a parameter instead, we won't have the problem of messing with immutable data and it will also enable us to pass an error object, alongside with an array object as data.

### PR DESCRIPTION
**Pull request by daffl**
Closes #64

### CODE REVIEW

**Comment by marshallswain** [comment]
@daffl is there some pending work, here?

**Comment by daffl** [comment]
Yes, there was some further discussion in https://github.com/feathersjs/feathers-errors/issues/64.

**Comment by ekryski** [comment]
Yeah, I might be missing something but I'm not sure if this is necessary. See my comment https://github.com/feathersjs/feathers-errors/issues/64#issuecomment-290826082

**Comment by eddyystop** [comment]
I've seen errors being passed back as an array for UI's which display all errors together, rather than field by field. Whereas errors being passed back as an object for Ui's that display errors by the field in question.

Is this what this issue is really about?

**Comment by daffl** [comment]
I'm not sure why this is controversial. The test shows an issue with a use case someone discovered and fixes it in a non-breaking way.

File: src/index.js [hunk 1]
```diff
@@ -29,16 +28,10 @@ class FeathersError extends Error {
...
-      // NOTE(EK): To make sure that we are not messing
-      // with immutable data, just make a copy.
-      // https://github.com/feathersjs/feathers-errors/issues/19
-      newData = Object.assign({}, data);
-
-      if (newData.errors) {
-        errors = newData.errors;
-        delete newData.errors;
-      }
+    if (data && data.errors) {
```
Comment (ekryski) [inline src/index.js]: There is a side effect here. If `data.errors` is `undefined` then the data object won't be cloned. If that data object is modified elsewhere it also be modified here.

The intention was that once the `data` object is passed in when creating a new Error that data object should be immutable.

**Comment by ekryski** [comment]
It is potentially breaking. I don't care whether data is an object or an array. It's fine if it handles both. This just needs to be tuned up a bit. I was merely pointing out that if the intention was for the `errors` object to be an array then that's already possible.

The `data` object is meant for any additional metadata beyond the `errors` object. For example, if you wanted to include the hook method or the user id.

File: src/index.js (hunk 1 above)
Comment (daniel-constantin) [inline src/index.js]: Can't we just clone the object using:

```
let clone = JSON.parse(JSON.stringify(data));
```
In contract to `Object.assign()` this works well with arrays.

**Comment by daffl** [comment]
Superseded by #75

**Pull request closed without merging**